packages/zoe/src/contracts/newSwap/getCurrentPrice.js

```diff
@@ -0,0 +1,260 @@
+// @ts-check
+
+import '../../../exported';
+
+import { assert, details as X } from '@agoric/assert';
+import { amountMath } from '@agoric/ertp';
+import { multiplyBy, makeRatio, natSafeMath } from '../../contractSupport';
+
+const { ceilDivide } = natSafeMath;
+const BASIS_POINTS = 10000;
+
+/**
+ * Build functions to calculate prices for multipoolAutoswap. Four methods are
+ * returned, as two complementary pairs. In one pair of methods the caller
+ * specifies the amount they will pay, and in the other they specify the amount
+ * they wish to receive. The two with shorter names (getOutputForGivenInput,
+ * getInputForGivenOutput) are consistent with the uniswap interface. They each
+ * return a single amount. The other two return { amountIn, centralAmount,
+ * amountOut }, which specifies the best exchange consistent with the request.
+ * centralAmount is omitted from these methods' results in the publicFacet.
+ *
+ * @param {(brand: Brand) => boolean} isSecondary
+ * @param {(brand: Brand) => boolean} isCentral
+ * @param {(brand: Brand) => Pool} getPool
+ * @param {Brand} centralBrand
+ * @param {bigint} poolFeeBP - fee (BP) to the pool in every swap.
+ * @param {bigint} protocolFeeBP - fee (BP) charged in RUN to every swap.
+ */
+
+export const makeGetCurrentPrice = (
+  isSecondary,
+  isCentral,
+  getPool,
+  centralBrand,
+  poolFeeBP,
+  protocolFeeBP,
+) => {
+  const protocolFeeRatio = makeRatio(protocolFeeBP, centralBrand, BASIS_POINTS);
+  const halfPoolFeeBP = ceilDivide(poolFeeBP, 2);
+
+  const getPriceGivenAvailableInput = (amountIn, brandOut) => {
+    const { brand: brandIn } = amountIn;
+
+    if (isCentral(brandIn) && isSecondary(brandOut)) {
+      // we'll charge the protocol fee before sending the remainder to the pool,
+      // so we need to subtract it from amountIn before getting a quote, and add
+      // it back before sending out the quote.
+      const prelimProtocolFee = multiplyBy(amountIn, protocolFeeRatio);
+      const poolAmountIn = amountMath.subtract(amountIn, prelimProtocolFee);
+      const price = getPool(brandOut).getPriceGivenAvailableInput(
+        poolAmountIn,
+        brandOut,
+        poolFeeBP,
+      );
+      const protocolFee = multiplyBy(price.amountIn, protocolFeeRatio);
+      const amountInFinal = amountMath.add(price.amountIn, protocolFee);
```

Comment:
Is there a reason we are adding the protocol fee here? This seems wrong to me.

Reply:
Ok, thanks for explaining. That makes sense to me. Could we add a comment for that?